### PR TITLE
Move code snippets to markdown

### LINKED ISSUE #1: Define code block in markdown instead of frontmatter
Currently code is defined in markdown frontmatter, which requires escaping and makes it impossible to see changes in the commit diffs. Look into using `gatsby-remark-prismjs` or see how it's done on https://github.com/gaearon/overreacted.io

### PR DESCRIPTION
**Pull request by bozheville**
Resolve #1 

## Snippets
Instead of having two single-string frontmatter fields, there are multiline `jsx` and `tsx` code snippets in the main md section. Their visibility is controlled by updating CSS rules, see CodeContent in `src/components/styled.js`

## Code highlighting
Use [gatsby-remark-vscode](https://www.gatsbyjs.com/plugins/gatsby-remark-vscode/) for code highlighting
Use [Tomorrow Night Eighties](https://marketplace.visualstudio.com/items?itemName=ms-vscode.Theme-TomorrowKit) theme. This is the most popular implementation of the theme

## Design update
"View in JS/TS" button moved to the bottom because it is defined out of mdx. It's is possible to create a button in a code snippet, but it should be done in a separate PR.

### CODE REVIEW

**Comment by bozheville** [comment]
@gragland,

I replaced Prismjs with VSCode highlighting because of styling for JSX. I could not find a proper theme for Prismjs, started to update it manually, but I think it is the wrong way. 

I use `Dark+ (default dark)` theme for now. I tried to use the same - Tomorrow - theme, but it is slightly different from what you have now on production. There is a list of themes to pick https://github.com/gerane/VSCodeThemes

**Comment by bozheville** [comment]
Hey @gragland,

This PR is finally not WIP anymore. See what's changed in an updated description. There are a lot of changes, but mostly it's updating of code snippets.

Sorry for the long waiting

**Comment by gragland** [comment]
@bozheville This is looking great! Thanks so much for your hard work on this.

Looks like the code is indented a bit. Any idea why this might be?
Screen Shot 2020-09-07 at 5 10 59 PM

**Comment by bozheville** [comment]
@gragland I found the reason, there is built-in padding for each row. But before I push the fix, I want to confirm that you can see the correct color theme. 

As I mentioned, I had to update a default theme, but these changes aren't applied on your screenshot. Here's how I see the same snippet. Or if you are happy with a default theme (as on your screenshot), I can revert a change

**Comment by gragland** [comment]
Ah yeah, sorry for the confusion. I had removed those changes to see what it looked like. I think it's fine to go with that default theme and not override the style.

**Comment by bozheville** [comment]
I updated styling. Reverted colors overriding and removed extra padding

**Comment by bozheville** [comment]
Hey @gragland,

Can I do something else on this PR?

**Comment by gragland** [comment]
@bozheville I'm noticing some highlighting issues for TS. Seems like the second const should not be red. Do you know if this is just a limitation of this particular theme or if we need to specify the syntax is TS somewhere?

Screen Shot 2020-09-18 at 8 39 29 AM

The other change would be moving the button above the code block using MDX, although like you said, we can do that in a separate pull-request.

**Comment by bozheville** [comment]
@gragland I changed `tsx` to `typescript` and now we have a better highlighting. Thanks for finding!

**Comment by gragland** [comment]
@bozheville Sorry for the delay on this! Looks great. If you have the time, feel free to open a new pull request for the button MDX. Once that's in I'll push these changes live.

**Comment by bozheville** [comment]
Hey @gragland, np. I'll open another PR for sure

File: src/pages/useAsync.md [hunk 1]
```diff
@@ -15,10 +15,156 @@ links:
...
+
+// An async function for testing our hook.
+// Will be successful 50% of the time.
+const myFunction = () => {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      const rnd = Math.random() * 10;
+      rnd <= 5
+        ? resolve('Submitted successfully uD83DuDE4C')
+        : reject('Oh no there was an error uD83DuDE1E');
```
Comment (kevin940726) [inline src/pages/useAsync.md:52]: Looks like there's an error when converting emoji to text here?

Comment (gragland) [reply src/pages/useAsync.md:52]: Thanks for catching this. Now fixed in https://github.com/gragland/usehooks/commit/0a85aa9a04ca40caf82e994d494a4c1492552f88

**Pull request merged**